### PR TITLE
[WIPTEST] Refactor freeipa ext auth tests

### PR DESCRIPTION
**Pull request by mshriver**
Refactoring of the authentication automation, including fixtures and tests.

PRT results should be clean, aside from what is skipped.

{{ pytest: cfme/tests/integration/test_cfme_auth.py cfme/tests/integration/test_aws_iam_auth_and_roles.py --long-running }}

### CODE REVIEW

File: cfme/utils/ext_auth.py [hunk 1]
```diff
@@ -30,47 +27,29 @@ def setup_external_auth_ipa(**data):
...
-        ipaserver_ssh.run_command('cp /etc/hosts /etc/hosts_bak')
-        ipaserver_ssh.run_command("sed -i -r '/^{}/d' /etc/hosts".format(appliance_address))
-        command = 'echo "{}\t{}" >> /etc/hosts'.format(appliance_address, appliance_fqdn)
-        ipaserver_ssh.run_command(command)
-    with current_appliance.ssh_client as ssh:
-        result = ssh.run_command('appliance_console_cli --host {}'.format(appliance_fqdn)).success
-        if not current_appliance.is_pod:
-            assert result
-        else:
+    # connect_kwargs = {
```
Comment (jamesooden) [inline cfme/utils/ext_auth.py]: Why leave this commented out when it will be there in previous versions of the file?

Comment (mshriver) [reply cfme/utils/ext_auth.py]: oh this is still very much a WIP :)

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by mshriver** [comment]
@jamesooden this is much closer to review if you'd like to check it out.

**Comment by dajoRH** [comment]
Lint report for commit d813310f4a8aad866273eb3519a62376d6cd27ee:

`cfme/tests/integration/test_aws_iam_auth_and_roles.py`:
- :red_circle: Line 7:1: **F401** *'cfme.utils.conf.auth_data' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f5e27431bcefc85939ab28f05526da7eca26e311:

`cfme/tests/integration/test_cfme_auth.py`:
- :bangbang: Line 13:1: **E999** *SyntaxError: can't assign to comparison*
- :bangbang: Line 100:5: **E303** *too many blank lines (2)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 1df5459fb818e53f61910244e73ca6a1400b768e:

`cfme/tests/integration/test_aws_iam_auth_and_roles.py`:
- :bangbang: Line 7:1: **E999** *SyntaxError: trailing comma not allowed without surrounding parentheses*

`cfme/tests/integration/test_cfme_auth.py`:
- :bangbang: Line 109:30: **E127** *continuation line over-indented for visual indent*
- :red_circle: Line 120:13: **F812** *list comprehension redefines 'user' from line 107*
- :red_circle: Line 149:5: **F841** *local variable 'credentials' is assigned to but never used*
- :red_circle: Line 149:40: **F821** *undefined name 'data'*
- :red_circle: Line 149:69: **F821** *undefined name 'data'*
- :red_circle: Line 150:5: **F841** *local variable 'role' is assigned to but never used*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 1185052fed10bbe4272ef3cef1fcc8b34f0c7030:

`cfme/tests/integration/test_aws_iam_auth_and_roles.py`:
- :bangbang: Line 7:1: **E999** *SyntaxError: trailing comma not allowed without surrounding parentheses*

`cfme/tests/integration/test_cfme_auth.py`:
- :bangbang: Line 109:30: **E127** *continuation line over-indented for visual indent*
- :red_circle: Line 120:13: **F812** *list comprehension redefines 'user' from line 107*
- :red_circle: Line 149:5: **F841** *local variable 'credentials' is assigned to but never used*
- :red_circle: Line 149:40: **F821** *undefined name 'data'*
- :red_circle: Line 149:69: **F821** *undefined name 'data'*
- :red_circle: Line 150:5: **F841** *local variable 'role' is assigned to but never used*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 150ce5670d8a5083f62f63c03ece0ba47a1518fb:

`cfme/fixtures/configure_auth_mode.py`:
- :red_circle: Line 1:1: **F401** *'copy.deepcopy' imported but unused*
- :red_circle: Line 6:1: **F401** *'cfme.utils.ext_auth.disable_external_auth_ipa' imported but unused*
- :red_circle: Line 6:1: **F401** *'cfme.utils.ext_auth.setup_external_auth_ipa' imported but unused*
- :heavy_exclamation_mark: Line 86:1: **W391** *blank line at end of file*

`cfme/utils/appliance/__init__.py`:
- :red_circle: Line 2161:45: **F821** *undefined name 'ldap_opts'*
- :red_circle: Line 2161:69: **F821** *undefined name 'ldap_opts'*
- :red_circle: Line 2163:60: **F821** *undefined name 'ldap_opts'*
- :red_circle: Line 2164:46: **F821** *undefined name 'ldap_opts'*
- :red_circle: Line 2167:41: **F821** *undefined name 'ldap_opts'*
- :red_circle: Line 2170:41: **F821** *undefined name 'ldap_opts'*
- :red_circle: Line 2179:13: **F402** *import 'conf' from line 28 shadowed by loop variable*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b0a5c51e2c4dda775aca5dadb8a858a2f2c358a9:

`cfme/utils/auth/__init__.py`:
- :bangbang: Line 11:1: **E302** *expected 2 blank lines, found 1*
- :heavy_exclamation_mark: Line 15:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 588a2d9d605e6833f5dab2fdf72a337f937f2bea:

`cfme/utils/auth/__init__.py`:
- :bangbang: Line 104:4: **E999** *IndentationError: expected an indented block*
- :bangbang: Line 104:5: **E303** *too many blank lines (2)*
- :bangbang: Line 104:5: **E112** *expected an indented block*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9833228f050e72aa76c905e308e995fde9c9ec34:

`cfme/fixtures/configure_auth_mode.py`:
- :red_circle: Line 5:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 6:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*
- :red_circle: Line 26:17: **F821** *undefined name 'auth_prov_data'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 983ba8ee846c31be1ccfbe825f3dba7d37e95ab1:

`cfme/fixtures/configure_auth_mode.py`:
- :red_circle: Line 5:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 6:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*
- :red_circle: Line 26:17: **F821** *undefined name 'auth_prov_data'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 7943815faefb0b16757a7ccd07635f3c25218ce3:

`cfme/fixtures/configure_auth_mode.py`:
- :red_circle: Line 5:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 6:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*
- :red_circle: Line 26:17: **F821** *undefined name 'auth_prov_data'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit baa05c097cc706450b81e3106af9ba703848fc3b:

`cfme/fixtures/configure_auth_mode.py`:
- :red_circle: Line 5:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 6:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*
- :red_circle: Line 26:17: **F821** *undefined name 'auth_prov_data'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3dc26f608ab0f8d3c115fd57e7a21d9075d22464:

`cfme/fixtures/configure_auth_mode.py`:
- :red_circle: Line 5:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 6:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*
- :red_circle: Line 26:17: **F821** *undefined name 'auth_prov_data'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 223a84771eb503329b2fae89320a93d11920d7fe:

`cfme/fixtures/configure_auth_mode.py`:
- :red_circle: Line 6:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 7:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*
- :red_circle: Line 27:17: **F821** *undefined name 'auth_prov_data'*

`cfme/tests/integration/test_cfme_auth.py`:
- :red_circle: Line 103:53: **F821** *undefined name 'auth_prov'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit eca937071e6a107900e1bead9a8b66c33a8828c7:

`cfme/fixtures/configure_auth_mode.py`:
- :red_circle: Line 6:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 7:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*
- :red_circle: Line 27:17: **F821** *undefined name 'auth_prov_data'*

`cfme/tests/integration/test_cfme_auth.py`:
- :red_circle: Line 103:53: **F821** *undefined name 'auth_prov'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ee2e267112caf977fef9a859e10a491548f4004b:

`cfme/configure/configuration/server_settings.py`:
- :red_circle: Line 12:1: **F401** *'cfme.utils.conf.credentials' imported but unused*

`cfme/fixtures/configure_auth_mode.py`:
- :red_circle: Line 6:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 7:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*
- :red_circle: Line 37:16: **F821** *undefined name 'auth'*

`cfme/tests/integration/test_cfme_auth.py`:
- :red_circle: Line 103:53: **F821** *undefined name 'auth_prov'*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c31b62d357b542280a5108e2a77add324dcdbfa5:

`cfme/configure/configuration/server_settings.py`:
- :red_circle: Line 12:1: **F401** *'cfme.utils.conf.credentials' imported but unused*

`cfme/fixtures/configure_auth_mode.py`:
- :red_circle: Line 6:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 7:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*
- :red_circle: Line 37:16: **F821** *undefined name 'auth'*

`cfme/tests/integration/test_cfme_auth.py`:
- :red_circle: Line 103:53: **F821** *undefined name 'auth_prov'*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 7f356d79b77941c1fcf4f755f801ef8251e709f8:

`cfme/configure/configuration/server_settings.py`:
- :red_circle: Line 12:1: **F401** *'cfme.utils.conf.credentials' imported but unused*

`cfme/fixtures/configure_auth_mode.py`:
- :red_circle: Line 6:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 7:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*

`cfme/tests/integration/test_cfme_auth.py`:
- :red_circle: Line 103:53: **F821** *undefined name 'auth_prov'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b33273eed2485a6180f3c083509ec2866703b4de:

`cfme/configure/configuration/server_settings.py`:
- :red_circle: Line 12:1: **F401** *'cfme.utils.conf.credentials' imported but unused*

`cfme/fixtures/configure_auth_mode.py`:
- :red_circle: Line 6:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 7:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*

`cfme/tests/integration/test_cfme_auth.py`:
- :red_circle: Line 103:53: **F821** *undefined name 'auth_prov'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c7f2810d3ba608c3d0549a4574f2d257622a990a:

`cfme/configure/configuration/server_settings.py`:
- :red_circle: Line 12:1: **F401** *'cfme.utils.conf.credentials' imported but unused*

`cfme/fixtures/authentication.py`:
- :red_circle: Line 4:1: **F401** *'cfme.utils.conf.credentials' imported but unused*
- :red_circle: Line 6:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 7:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6b27130fedebd6b88f85276b13570a9d0c54f7c6:

`cfme/configure/configuration/server_settings.py`:
- :red_circle: Line 12:1: **F401** *'cfme.utils.conf.credentials' imported but unused*

`cfme/fixtures/authentication.py`:
- :red_circle: Line 4:1: **F401** *'cfme.utils.conf.credentials' imported but unused*
- :red_circle: Line 6:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 7:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 39a638f9d6e4b42cf1636ab21d528ce3f2b491ad:

`cfme/configure/configuration/server_settings.py`:
- :red_circle: Line 12:1: **F401** *'cfme.utils.conf.credentials' imported but unused*

`cfme/fixtures/authentication.py`:
- :red_circle: Line 5:1: **F401** *'cfme.utils.conf.credentials' imported but unused*
- :red_circle: Line 7:1: **F401** *'cfme.utils.path.conf_path' imported but unused*
- :red_circle: Line 8:1: **F401** *'cfme.utils.ssh.SSHClient' imported but unused*

`cfme/tests/integration/test_cfme_auth.py`:
- :red_circle: Line 6:1: **F401** *'cfme.configure.configuration.server_settings.AUTH_MODES' imported but unused*
- :red_circle: Line 11:1: **F401** *'cfme.utils.blockers.BZ' imported but unused*
- :bangbang: Line 38:9: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 65:38: **E231** *missing whitespace after ':'*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by mshriver** [comment]
Going to reopen to cleanup the history here a bit, and since the scope changed so much from the original intent.

**Pull request closed without merging**